compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1593,7 +1593,8 @@ impl<'tcx> Place<'tcx> {
     }
 
     /// If MirPhase >= Derefered and if projection contains Deref,
-    /// It's guaranteed to be in the first place
+    /// It's guaranteed to be in the first place, this functions checks
+    /// both of these conditions and must only be called while MirPhase >= Derefered
     pub fn has_deref(&self) -> bool {
```

Comment:
The name still needs changing IMO. This and `is_indirect` are implementing the same check so that should have very similar names.

What about `is_indiect_first_projection` or `is_indirect_mir_derefered` or so?

Reply:
Changed it to `is_indirect_first_projection` I think it suits better